UnityProject/Assets/Scripts/Objects/Other/MouseTrap.cs

```diff
@@ -99,26 +99,34 @@ private void UpdateTrapVisual()
 			trapPreview.SetSpriteSO(sprite.GetCurrentSpriteSO());
 		}
 
-		public override void OnStep(GameObject eventData)
+		public override bool CanPlayerStep(PlayerScript playerScript)
+		{
+			return playerScript.IsGhost == false;
+		}
+
+		public override void OnPlayerStep(PlayerScript playerScript)
 		{
 			if (IsArmed == false) return;
 			if(trapInSnare) TriggerTrap();
-			//a mouse trap must kill mice, duh
-			//TODO : IEnterable is designed for players only so mice can't trigger this :(
-			if (eventData.TryGetComponent<MouseAI>(out var mouse))
-			{
-				mouse.health.Death();
-				return;
-			}
-			base.OnStep(eventData);
+
+			base.OnPlayerStep(playerScript);
 			isArmed = false;
 		}
 
-		public override bool WillStep(GameObject eventData)
+		public override bool CanObjectEnter(GameObject eventData)
 		{
-			if (eventData.gameObject.TryGetComponent<LivingHealthMasterBase>(out var _)) return true;
-			if (eventData.gameObject.TryGetComponent<MouseAI>(out var _mouse)) return true;
-			return false;
+			//Damage mice
+			return eventData.HasComponent<MouseAI>();
+		}
+
+		public override void OnObjectEnter(GameObject eventData)
+		{
+			if (IsArmed == false) return;
+			if(trapInSnare) TriggerTrap();
+
+			//A mouse trap must kill mice, duh
+			eventData.GetComponent<MouseAI>().health.Death();
```

Comment:
missing the null check

Reply:
> missing the null check

Checked in the CanObjectEnter

Reply:
ah, only mice can trigger the effect? Aight